plugins/org.jkiss.dbeaver.ext.sqlite/src/org/jkiss/dbeaver/ext/sqlite/edit/SQLiteTableManager.java

```diff
@@ -78,6 +81,16 @@ public void renameObject(@NotNull DBECommandContext commandContext, @NotNull Gen
         processObjectRename(commandContext, object, options, newName);
     }
 
+    @Override
+    protected boolean isIncludeConstraintInDDL(DBRProgressMonitor monitor, DBSEntityConstraint constraint) {
+        if (constraint.getConstraintType() == DBSEntityConstraintType.PRIMARY_KEY && constraint instanceof GenericUniqueKey) {
+            if (((GenericUniqueKey)constraint).getAttributeReferences(monitor).size() == 1 && ((GenericUniqueKey)constraint).getAttributeReferences(monitor).get(0).getAttribute().isAutoIncrement()) {
```

Comment:
Just a nitpick:
```suggestion
        if (constraint.getConstraintType() == DBSEntityConstraintType.PRIMARY_KEY && constraint instanceof GenericUniqueKey key) {
            if (key.getAttributeReferences(monitor).size() == 1 && key.getAttributeReferences(monitor).get(0).getAttribute().isAutoIncrement()) {
```
Additionally, I'd extract the result of `getAttributeReferences` into a separate variable to improve readability.

Reply:
Thanks for review and advice, and sorry for the code, I'm a very casual java developer ;)
I just pushed a new version which takes into this comments account.